compiler/src/compilers/compiler.rs

```diff
@@ -0,0 +1,7 @@
+use super::super::data::input_data::InputData;
+use super::super::data::output_data::OutputData;
+
+
+pub trait Compiler {
+    fn compile(&self, input_data: &InputData) -> Result<OutputData, &'static str>;
```

Comment:
Seems like this call can block for a considerable amount of time, can this function be `async`?